### PR TITLE
Improvements 10 01 24c

### PR DESCRIPTION
**Pull request by willkessler**
many improvements from week of oct 1

### CODE REVIEW

File: src/lib/components/ActionItems.svelte [hunk 1]
```diff
@@ -187,132 +196,150 @@
...
     </Table>
-
-  {/if}
-</div>
+  {:else} <!-- individual action item, shows in right panels -->
+    <div class=" mb-4 flex justify-between">
+      <h2 class="text-xl font-bold text-customGray uppercase">Action Details</h2>
+      <div on:click={() => { hideRightPanelCb(); }}
+        class="border rounded-xs px-2 cursor-pointer bg-customGray/10 hover: hover:bg-customGray/20">
+        X
```
Comment (amkoo7) [inline src/lib/components/ActionItems.svelte:294]: This isn't a svelte component. How come we're not just using the standard close button (examples in the drawer components) or a ... component if you foresee more functionality such as edit, close, etc.?

File: src/lib/components/ActionItems.svelte [hunk 2]
```diff
@@ -187,132 +196,150 @@
...
+    {#if (actionItemId !== null) }
+      <div class="grid grid-cols-2 gap-y-2 gap-x-4 mt-4">
+        <div class="font-semibold">Action:</div>
+        <div class="text-customGray">{getActionItemById(actionItemId).name}</div>
+
+        <div class="font-semibold">Taken:</div>
+        <div class="text-customGray">{getActionItemById(actionItemId).eventDate}</div>
+
+        <div class="font-semibold">Action type:</div>
+        <CustomBadge
```
Comment (amkoo7) [inline src/lib/components/ActionItems.svelte:306]: Minor: I think on hover, this and other badges should cause the cursor to a selector

File: src/lib/components/ActionItems.svelte (hunk 1 above)
Comment (amkoo7) [reply src/lib/components/ActionItems.svelte:294]: Actually, you use the svelte component in the add action item modal

File: src/lib/components/ActionItemsTable.svelte [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+         canAddMoreOptions: false
+       },
+       iconSettings: {
+         reusableIconName: 'select',
+       },
+       label: {
+         options: [
+           {text: 'Maintenance', backgroundColor: '#f1fecd'},
+           {text: 'Record', backgroundColor: '#e1ff8f'},
+           {text: 'Other/Insurance', backgroundColor: '#cdfef7'},
```
Comment (amkoo7) [inline src/lib/components/ActionItemsTable.svelte:33]: Minor: Should we just make this two separate items: Other, Insurance?

File: src/lib/components/ActionItemsTable2.svelte
Comment (amkoo7) [inline src/lib/components/ActionItemsTable2.svelte:1]: @willkessler there are two ActionItemsTable 😭?

File: src/lib/data/activityData.js [hunk 4]
```diff
@@ -131,7 +163,77 @@ export const incidents = [
     driverId: 1922,
     vehicleId: 4396,
     level: 'State',
-    description:'Tom called this one in.',
+    description:'Thom called this one in after feeling some rear wobble on Highway 64.',
```
Comment (amkoo7) [inline src/lib/data/activityData.js:166]: Nit: 'Tom' or 'Thomas'

File: src/lib/data/activityData.js [hunk 5]
```diff
@@ -131,7 +163,77 @@ export const incidents = [
...
+    type: 'Maintenance',
+    category: 'Maintenance',
+    status: 'Closed',
+    occurrenceDate: 'Feb 13, 2024', 
+    dueDate: 'Jun 13, 2024',
+    title: 'Spark plugs replacement needed',
+    driverId: 1922,
+    vehicleId: 4396,
+    level: 'State',
+    description:'Thom says the weigh station mechanic explained weak power on this vehicle is due to old spark plugs.',
```
Comment (amkoo7) [inline src/lib/data/activityData.js:194]: Same minor nit

**Review by amkoo7 — APPROVED** [review]
@willkessler left a few comments

File: src/lib/components/ActionItems.svelte (hunk 2 above)
Comment (willkessler) [reply src/lib/components/ActionItems.svelte:306]: you mean `cursor:pointer` ?  customBadges already have this:

```
 $: badgeClasses = `px-2 py-1.5 rounded-[6px] min-w-32 min-h-9 text-${colorAndIcon.color}-${textSaturation} bg-${colorAndIcon.color}-${bgSaturation} cursor-pointer`;
```

File: src/lib/components/ActionItems.svelte (hunk 1 above)
Comment (willkessler) [reply src/lib/components/ActionItems.svelte:294]: not using drawer so... their flowbite X is not separable. rolled my own but i can make sure the X matches the one in the drawer for consistency.

Comment (willkessler) [reply src/lib/components/ActionItems.svelte]: @amkoo7 i made the x match the ones in the flowbite drawer in the "d" branch. that's merged into this one now.

File: src/lib/components/ActionItemsTable2.svelte
Comment (willkessler) [reply src/lib/components/ActionItemsTable2.svelte:1]: yeah, will clean that up later today

File: src/lib/components/ActionItemsTable.svelte (hunk 3 above)
Comment (willkessler) [reply src/lib/components/ActionItemsTable.svelte:33]: sure

File: src/lib/data/activityData.js (hunk 5 above)
Comment (willkessler) [reply src/lib/data/activityData.js:208]: you never seen the spelling "Thom"? yeah, it's a thing

**Comment by willkessler** [comment]
LG

**Pull request merged**